infra/src/main/kotlin/fi/fta/geoviite/infra/linking/SwitchLinkingService.kt

```diff
@@ -1585,10 +1585,14 @@ private fun getLocationTrackChangesFromLinkingSwitch(
         }
         .map { (_, track) -> track }
     return LocationTrackChangesFromLinkingSwitch(
-        onlyDelinked, onlyTopoLinkChanged.values.toList(), alignmentLinksChanged.values.toList()
+        draft(onlyDelinked), draft(onlyTopoLinkChanged.values.toList()), draft(alignmentLinksChanged.values.toList())
```

Comment:
Tämä oletettavasti on käyty läpi dailyissä jne. ja todettu OK:ksi ratkaisuksi, mutta tämä tuntuu silti jotenkin häkkyrältä. Intuitioni sanoisi että jos validointilogiikka riippuu siitä onko asiat drafteja vai ei ja sen vuoksi me vaan rubberstampataan draftiksi kaikki tämän funktion läpi kulkeva tavara jotta saadaan validaatio toimimaan, niin eikös oikeampi ratkaisu olisi muuttaa sitä validaatiota niin, että se ei perustu draftiuteen? Vai onko nämä sijaintiraiteita, jotka olisi pitänyt merkata draftiksi muutenkin (ja jotka syystä tai toisesta eivät ole merkkautuneet sellaisiksi jo aiemmin?) Jos näin, niin onko tämä oikea paikka merkata ne sellaisiksi, vai pitäisikö ne draftata jossain muualla?

Reply:
Joo tämä näyttää kyllä aika oudolta. Voisiko tuo itse asiassa olla vain sitä että validoinnissa oli bugi jonka seurauksena se ei validoinut official-kamaa oikein? Siis tämä: https://github.com/finnishtransportagency/geoviite/pull/1123#pullrequestreview-1905443672